ducktape/cluster/remoteaccount.py

```diff
@@ -638,14 +638,36 @@ def open(self, path, mode='r'):
     @check_ssh
     def create_file(self, path, contents):
         """Create file at path, with the given contents.
-
         If the path already exists, it will be overwritten.
         """
         # TODO: what should semantics be if path exists? what actually happens if it already exists?
         # TODO: what happens if the base part of the path does not exist?
-
-        with self.sftp_client.open(path, "w") as f:
-            f.write(contents)
+        try:
+            self._log(logging.DEBUG,
+                      f"Let's create or overwrite file at: {path}")
+            
+            dir_path = os.path.dirname(path)
+            if dir_path:
+                self._log(logging.DEBUG,
+                          f"Checking if directory exists: {dir_path}")
+                os.makedirs(dir_path, exist_ok=True)
+                stat = os.statvfs(dir_path)
+                free_space = stat.f_frsize * stat.f_bavail
+                self._log(logging.DEBUG,
+                          f"File size: {len(contents)} bytes")
+                self._log(logging.DEBUG,
+                          f"Available disk space in {dir_path}: {free_space} bytes")
```

Comment:
Even if we fix this to be on the remote node, I am not sure we want/need all the logs default? I would think we just need to retroactively add some debug logs.

So if this fails, check if the node is still reachable (via ping?), then see if you can connect over SSH? then we can check remote information after the fact to see if that is the issue.

We do these operations all the time in tests so it will be a ton of extra noise to have this.